### PR TITLE
Bug Fixes and cleanup

### LINKED ISSUE #25: Disconnected ready button
When a player disconnects, ready status is reset HOWEVER the button clickable state is not properly reset to match.

### PR DESCRIPTION
**Pull request by BrandonSweatman**
Cleaned up some code.

-Added some error protection in a few scripts to check for the existence of singletons before assignments.
-Attempted to fix a bug where a disconnected client that already clicked Ready, would rejoin and not be able to click ready button( many tested situations saw this issue fixed, however I saw one instance I could not reproduce where the button was grey out when it should not be.)
-Fixed the bug where only 1 player was being showed in the ActiveGameMenu. The player button was there. but all buttons were stacked. fixed by adding a grid to the view container.

### CODE REVIEW

**Comment by BrandonSweatman** [comment]
I took some time to address those two main bugs we encountered during the review.

I would like to do some UI refactoring while I'm at it. I agree with you that the UI should be more uniform. my question to you is Which direction should I go.  Should the Active Game Menu behave like the end game menu? or visa versa?

Edit: 
Should the player choices occur in a popup list, or be active buttons without opening a list?

**Comment by AvinZarlez** [comment]
I think the "scrollable list" is way to go from a UX experience, so probably a popup. This would also allow moire screen real estate to be able to tell the player what they need to do. Then once they hit the popup button, seeing the large scrollable list of colors will be clear that these are "other players" and the giant box at the top of the screen is "their" color. Help avoid confusion of people wondering what color they are.

File: .gitignore [hunk 1]
```diff
@@ -3,3 +3,9 @@ Dance Together/Assets/Private/
 Dance Together/Assets/Demigiant/
 *.wav
 *.wav.meta
+Dance Together/Assets/Scripts/Controllers/GameManager.cs
```
Comment (AvinZarlez) [inline .gitignore:6]: Why is this being ignored?

**Review by AvinZarlez — CHANGES_REQUESTED** [review]
Seems good, but is there a reason to add to the gitignore? I understand why you'd add locally if those files exist and you don't want to accidentally add them, but those .gitignore changes don't need to be committed.

File: .gitignore (hunk 1 above)
Comment (BrandonSweatman) [reply .gitignore:6]: My appologies. That was legacy ignore stuff that i didnt intend on commiting.

Comment (AvinZarlez) [reply .gitignore:6]: I figured. I merged and removed them :)

**Pull request merged**